sips/fork_support.md

```diff
@@ -20,56 +20,259 @@ Client implementation should calculate when a new duty starts to decide if it st
+	ShifuNetworkID   = NetworkID(0x2)
+	JatoNetworkID    = NetworkID(0x3)
 )
 
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
 var (
-    GenesisMainnet = DomainType{0x0, 0x0, GenesisChain, 0x0}
-    PrimusTestnet  = DomainType{0x0, 0x0, PrimusChain, 0x0}
-    ShifuTestnet   = DomainType{0x0, 0x0, ShifuChain, 0x0}
-    ShifuV2Testnet = DomainType{0x0, 0x0, ShifuChain, 0x1}
-    JatoTestnet    = DomainType{0x0, 0x0, JatoChain, 0x1}
+	GenesisMainnet = DomainType{0x0, 0x0, byte(MainnetNetworkID), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, byte(PrimusNetworkID), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x0}
+	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
 )
 
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
 type ForkData struct {
-    Epoch  phase0.Epoch
-    Domain DomainType
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType
+}
+
+func (domainType DomainType) GetNetworkID() NetworkID {
+	return NetworkID(domainType[2])
 }
 
-func (domainType DomainType) GetChain() byte {
-    return domainType[2]
+func (networkID NetworkID) GetForksData() []*ForkData {
+	switch networkID {
+	case MainnetNetworkID:
+		return mainnetForks()
+	default:
+		return []*ForkData{
+			{
+				Epoch:  0,
+				Domain: DomainType{0x0, 0x0, byte(networkID), 0x0},
+			},
+		}
+	}
 }
 
-func (domainType DomainType) GetForksData() []*ForkData {
-    switch domainType.GetChain() {
-    case GenesisChain:
-        return genesisForks()
-    default:
-        return []*ForkData{}
-    }
+// mainnetForks returns all forks for the mainnet chain
+func mainnetForks() []*ForkData {
+	return []*ForkData{
+		{
+			Epoch:  0,
+			Domain: GenesisMainnet,
+		},
+	}
 }
 
-func genesisForks() []*ForkData {
-    return []*ForkData{
-        {
-            Epoch:  0,
-            Domain: GenesisMainnet,
-        },
-    }
+func (networkID NetworkID) DefaultFork() *ForkData {
+	return networkID.GetForksData()[0]
 }
 
+// GetCurrentFork returns the ForkData with highest Epoch smaller or equal to "epoch"
+func (networkID NetworkID) GetCurrentFork(epoch phase0.Epoch) (*ForkData, error) {
+	// Get list of forks
+	forks := networkID.GetForksData()
+
+	// If empty, raise error
+	if len(forks) == 0 {
+		return nil, errors.New("GetCurrentFork: fork list by GetForksData is empty.")
+	}
+
+	current_fork := forks[0]
+	for _, fork := range forks {
+		if fork.Epoch <= epoch && fork.Epoch > current_fork.Epoch {
+			current_fork = fork
+		}
+	}
```

Comment:
```suggestion
	var current_fork *ForkData
	for _, fork := range forks {
		if fork.Epoch <= epoch {
			current_fork = fork
		}
	}
```

Reply:
So the list of forks is guaranteed to ordered, right?

Reply:
> So the list of forks is guaranteed to ordered, right?

right, we can do the same as beacon API `/eth/v1/config/fork_schedule`
```
{
    "data": [
        {
            "previous_version": "0x00001020",
            "current_version": "0x00001020",
            "epoch": "0"
        },
        {
            "previous_version": "0x00001020",
            "current_version": "0x01001020",
            "epoch": "36660"
        },
        {
            "previous_version": "0x01001020",
            "current_version": "0x02001020",
            "epoch": "112260"
        },
        {
            "previous_version": "0x02001020",
            "current_version": "0x03001020",
            "epoch": "162304"
        }
    ]
}
```

but not sure why we need the previous version

Reply:
Indeed, I don't see why we should use either. In case we need the previous fork of a certain epoch, we can add a function later to it. What do you think?

Reply:
Sure